update select menu appends

gitdallas opened the pull request:
https://issues.redhat.com/browse/COST-2617

ended up using inline in cases where the menu for sure fits between edges of modal and doesn't go over header/footer of wizard with the new height restriction (scrollable), and parent for cases where it goes off the edge of the modal. both ways work with keyboard and a11y (being close in dom).

the 350 max height currently only affects the currency select, and gives it most of the usable space between wizard header/footer while cutting off at half an item so it's more obvious that scrolling is possible.

dlabrecq commented on src/pages/costModels/components/inputs/selector.tsx:
```diff
@@ -94,6 +94,7 @@ const SelectorBase: React.FunctionComponent<SelectorProps> = ({
     >
       <Select
         id={id}
+        maxHeight={350}
```
There is `maxHeight` and other variables stored in wizard.styles. Can we add this value there?

gitdallas replied:
I was applying the maxHeight to any use of the select dialog. Right now wizard styles are being used by the generalInformation component only. They are also always being used as real styles (passed into `style` attributes on basic HTML elements), this value is a custom PF prop that gets handled by PF rather than as an actual style attribute.

If desired, I could create a maxHeight prop on our Selector component that we then pass into the PF Select component.. and just pass it to our Selector component from generalInformation with a `maxHeight={styles.selector.maxHeight`.

dlabrecq replied:
Yes, `maxHeight={styles.selector.maxHeight}` is what I was looking for. We typically place all style properties, like height in a `styles` file -- even though PatternFly may treat this one differently. Easier to find properties and / or reuse them when located together.

I agree that a `maxHeight` prop would be best, so we can use it only when necessary.

dlabrecq reviewed (commented):
Looks good, just one small nit.

I ran this locally, but can you please add a screenshot to the PR. Just in case Natalie or QE wants to review?

gitdallas commented:
https://user-images.githubusercontent.com/5322142/168086854-c154ea38-881d-4438-93de-eb02cc1b4d7f.mov

nlcwong commented:
LGTM

gitdallas commented:
added component ids in for qa testers, added aria labels on the generalInfo select toggles, created a maxHeight property in our selector to pass to PF select and setting to 350 for general info items

dlabrecq reviewed (approved):
LGTM
